src/EnergyPlus/UnitarySystem.cc

```diff
@@ -13268,22 +13268,24 @@ namespace UnitarySystems {
                                                                                                       singleMode);
                             if ((state.dataLoopNodes->Node(OutletNode).HumRat - DesOutHumRat) < Acc) break;
                         }
+                        // make sure outlet HumRat is below set point before calling SolveRoot
+                        if (state.dataLoopNodes->Node(OutletNode).HumRat < DesOutHumRat) {
+                            std::array<Real64, 8> Par2 = {
+                                double(this->m_CoolingCoilIndex),
```

Comment:
Also, not part of this PR, but at some point these should be converted to C++ lambdas. See [this issue](https://github.com/NREL/EnergyPlus/issues/9110) and [this tutorial on lambdas](https://github.com/NREL/EnergyPlus/wiki/Lambdas:-When-and-How-to-Use-Them).